### PR TITLE
Updated readme, to reflect new changes

### PR DESCRIPTION
**Pull request by ajmas**
Process changes, so we needed to update the documentation. Also, taking the opportunity to make it clearer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
+content, we need to use gems (Ruby modules) that are not
+supported by GitHub's gh-pages Jekyll scripts. For this
+reason we run jekyll with the help of [Travis](https://travis-ci.com/).
 
-`gem install jekyll-sitemap`
+Notes to Mentors
+----------------
+
+If you are adding yourself or removing yourself as a 
+mentor, you should create a PR request, first having modified
```
Comment (RichardLitt) [inline README.md:16]: Against which branch?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,48 @@
...
+Note, that the `script/deploy` is intended to be used
+by Travis and should not be run locally. When run it
+will do all the magic necessary to take the generated
+site and push it to the gh-pages branch.
+
+Pull Request
+------------
+
+Remember when making your pull request (PR) to do it against
+the [gh-pages-src](https://github.com/nodeschool/montreal/tree/gh-pages-src) branch. Doing anything else will most definitely
```
Comment (RichardLitt) [inline README.md:41]: Why do we have other branches? Can we just have the one, or at the least set it as the default?

File: README.md (hunk 1 above)
Comment (ajmas) [reply README.md:16]: That was stated in a section below, but we can put it in this section.

File: README.md (hunk 2 above)
Comment (ajmas) [reply README.md:41]: Not sure why we have both `master` and `gh-pages`? As for `gh-pages-src`, this is where we are publishing from to `gh-pages`, as part of the Travis CI process. I wanted to do the changes incrementally, until I had feedback from the relevant people.

We could change things up, such that `master` and `gh-pages-src` are merged into one branch and that we publish to `gh-pages`. I don't want to do this until I have an okay from @pcuci 

GitHub does not support i18n at this point. There is the following open issue tracking this: https://github.com/github/pages-gem/issues/401

File: README.md (hunk 1 above)
Comment (RichardLitt) [reply README.md:16]: Noticed that. This is fine.

File: README.md (hunk 2 above)
Comment (RichardLitt) [reply README.md:41]: Then let's remove `master`, or, better, remove `gh-pages-src` and rename it to `master`, and just have people push all commits against master, which would then generate `gh-pages` automatically. I think this is a more common flow. @pcuci sound good?

**Comment by pcuci** [comment]
The NodeSchool.io/montreal site relies on a `gh-pages*` branch, IIRC

Do as you please, no need to loop me in, all is revertable :-)

File: README.md (hunk 2 above)
Comment (RichardLitt) [reply README.md:41]: @ajmas Can we go with this flow, then?

**Comment by ajmas** [comment]
Based on a conversation with Richard we will make the master branch the source branch and gh-pages the published branch. This may also make it more convenient for new mentors, since this will mean only one branch needs updating.

File: README.md (hunk 2 above)
Comment (RichardLitt) [reply README.md:41]: Needs to be updated.

**Comment by ajmas** [comment]
Closing PR, since we need to do this against another branch. Note README.md updated in master, let me know if there any changes that need to be made there or you can do changes yourself.

**Comment by RichardLitt** [comment]
I'm not seeing another PR. This stuff is all still needed in the README, innit?

**Pull request closed without merging**